cocotb/share/include/gpi_logging.h

```diff
@@ -37,39 +37,183 @@
 #define GPILOG_EXPORT COCOTB_IMPORT
 #endif
 
+#include <cstdarg>
+
 #ifdef __cplusplus
-# define EXTERN_C_START extern "C" {
-# define EXTERN_C_END }
-#else
-# define EXTERN_C_START
-# define EXTERN_C_END
+extern "C" {
 #endif
 
-EXTERN_C_START
-
+/** Named logging level
+ *
+ *  Only logs at one of these log level will print a level name.
```

Comment:
```suggestion
 *  Only logs at one of these log level will print a level name with the "native" logger
```
This statement is false for the python logger.